core/domain/stats_services.py

```diff
@@ -294,9 +331,14 @@ def advance_version_of_exp_stats(
 
     new_state_name_stats_mapping = {}
 
+    if exp_versions_diff is None:
+        raise Exception(
+            'ExplorationVersionsDiff cannot be None when the change is'
```

Comment:
I don't think I understand this. Isn't the revert case handled in the block of code above?

Reply:
> Isn't the revert case handled in the block of code above?

Yes we are handling the revert case above, but that was only for the case where `revert_to_version` is not none and `exp_versions_diff` is none.

Becasue it is not a private method so I think this can be called in other places ( may be in the future ) and there we provide both `revert_to_version and exp_versions_diff` as none. So in this, I think case error makes sense, right?

Reply:
I don't see the condition on exp_versions_diff above. When reading the code I think the above code (lines 316-330) executes and returns if revert_to_version is true-y. Is that not correct?

Reply:
> I think the above code (lines 316-330) executes and returns if revert_to_version is true-y. Is that not correct?

Yes, it is correct..

Error message is changed from `ExplorationVersionsDiff cannot be None when the change is a revert.` to `ExplorationVersionsDiff cannot be None when the change is not a revert.`